test/functional/sharding_read_preference_tests.js

```diff
@@ -90,35 +90,17 @@ describe('Sharding (Read Preference)', function() {
         collection.insertOne({ test: 1 }, { w: 'majority', wtimeout: 10000 }, function(err) {
           expect(err).to.not.exist;
 
-          // Set debug level for the driver
-          Logger.setLevel('debug');
+          const findOneFunction = function() {
+            collection.findOne({ test: 1 }, { readPreference: new ReadPreference('notsupported') });
```

Comment:
This is no longer testing what was originally being test. The original test is showing that when we send an invalid readPreference to the server, we get a `MongoError` back. Now, we're breaking on an invalid ReadPreference before we send anything to the server. As long as we have coverage showing that `new ReadPreference('notsupported')` throws an error, we could probably delete this entire test.

Reply:
Good point. This is covered by the read preference [test](https://github.com/mongodb/node-mongodb-native/blob/2710a073ff5e09914145638ac33c0503f086e324/test/functional/readpreference_tests.js#L551) I added already.